oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticConnection.java

```diff
@@ -108,11 +123,41 @@ public RestHighLevelClient getClient() {
                         Header[] headers = new Header[]{new BasicHeader("Authorization", "ApiKey " + apiKeyAuth)};
                         builder.setDefaultHeaders(headers);
                     }
-                    client = new RestHighLevelClient(builder);
+                    RestClient httpClient = builder.build();
+                    hlClient = new RestHighLevelClientBuilder(httpClient).setApiCompatibilityMode(true).build();
+
+                    ElasticsearchTransport transport = new RestClientTransport(
+                            httpClient, new JacksonJsonpMapper());
+                    esClient = new ElasticsearchClient(transport);
+                    esAsyncClient = new ElasticsearchAsyncClient(transport);
```

Comment:
Could there be a case where esClient could be null but not esAsyncClient ?

Reply:
no, this cannot happen